src/neptune/metadata_containers/metadata_container.py

```diff
@@ -256,8 +256,8 @@ def _handle_fork_in_child(self):
                     flush_period=self._flush_period,
                     queue=self._signals_queue,
                 ),
-                post_trigger_side_effect=self._op_processor.start,
             )
+            self._op_processor.set_post_trigger_side_effect(self._op_processor.start)
```

Comment:
Isn't it simpler to set it as a part of the constructor?

Reply:
its actually a bit tricky to do it in a constructor. Previously we had
```
# self._op_processor was already defined
self._op_processor = LazyOperationProcessorWrapper(
                operation_processor_getter=partial(
                    get_operation_processor,
                    mode=self._mode,
                    container_id=self._id,
                    container_type=self.container_type,
                    backend=self._backend,
                    lock=self._lock,
                    flush_period=self._flush_period,
                    queue=self._signals_queue,
                ),
                post_trigger_side_effect=self._op_processor.start,
            )
```

so in line `post_trigger_side_effect=self._op_processor.start`
was `self._op_processor` was pointing to already defined `op_processor` like `AsyncOperationProcessor`.
which was also used by parent process (fork model) and therefore both processes would reference the same `consumer` thread which caused errors down the line.

Now `self._op_processor` points to `LazyOperationProcessorWrapper` when triggered will create new `AsyncOperationProcessor` instance with new `consumer` thread in fork child

Reply:
You don't have to provide it as a parameter. You can set up it internally in the `LazyOperationProcessor` as:
`self._post_trigger_side_effect=self.start`

Reply:
I know that you cannot provide it as a parameter because you don't have an instance created :wink:

Reply:
In theory I could define `post_trigger_side_effect` as function that takes one argument - `self` of `LazyOperationProcessorWrapper` and work my way around that

Reply:
I mean that you can set this property internally as a part of a constructor of `LazyOperationProcessorWrapper` like: `self._post_trigger_side_effect=self.start`. Think that it needs to be always called and it's an "atomic" operation that you have to initialize OperationProcessor and once it's created, run `start` on init. I've prepared some suggestions here: https://github.com/neptune-ai/neptune-client/pull/1705